riscv/mmu.h

```diff
@@ -141,10 +141,12 @@ class mmu_t
   // template for functions that perform an atomic memory operation
   template<typename T, typename op>
   T amo(reg_t addr, op f) {
+    check_triggers(triggers::OPERATION_AMO, addr);
     convert_load_traps_to_store_traps({
       store_slow_path(addr, sizeof(T), nullptr, 0, false, true);
       auto lhs = load<T>(addr);
```

Comment:
Isn't it possible that this `load` will accidentally hit the data trigger, therefore causing an unintended side effect without performing the store?

I think you might be able to call `load_slow_path_intrapage` then `from_target` to avoid this.  Not sure if there is a cleaner way.